### PR TITLE
load_earth_relief: Add the support of data source 'GEBCOSI'

### PR DESCRIPTION
**Pull request by willschlitzer**
This pull request adds the GEBCOSI data source to `load_earth_relief`

Addresses #1786

**Reminders**

- [ ] Run `make format` and `make check` to make sure the code follows the style guide.
- [ ] Add tests for new features or tests that would have caught the bug that you're fixing.
- [ ] Add new public functions/methods/classes to `doc/api/index.rst`.
- [ ] Write detailed docstrings for all functions/methods.
- [ ] If wrapping a new module, open a 'Wrap new GMT module' issue and submit reasonably-sized PRs.
- [ ] If adding new functionality, add an example to docstrings or tutorials.

**Slash Commands**

You can write slash commands (`/command`) in the first line of a comment to perform
specific operations. Supported slash commands are:

- `/format`: automatically format and lint the code
- `/test-gmt-dev`: run full tests on the latest GMT development version

### CODE REVIEW

**Comment by seisman** [comment]
This PR has some conflicts with the main branch. 

@willschlitzer Do you want to fix these conflicts and get the PR merged before PR #2200? Or we can focus on PR #2200 first and add the "GEBCOSI" data source later.

**Comment by willschlitzer** [comment]
> This PR has some conflicts with the main branch.
> 
> @willschlitzer Do you want to fix these conflicts and get the PR merged before PR #2200? Or we can focus on PR #2200 first and add the "GEBCOSI" data source later.

I think it makes sense to finish #2200 first. But if you want to add the named tuples in to the current PR, as opposed to a later one, I would prefer to finish this first, since I suspect it would be some time before I could merge this.

File: pygmt/helpers/testing.py [hunk 1]
```diff
@@ -156,6 +156,8 @@ def download_test_data():
     datasets = [
         # Earth relief grids
         "@earth_gebco_01d_g",
+        "@earth_gebcosi_01d_g",
+        "earth_gebcosi_15m_p",
```
Comment (seisman) [inline pygmt/helpers/testing.py:160]:
```suggestion
        "@earth_gebcosi_15m_p",
```

**Review by seisman — APPROVED** [review]
Looks good to me except for the above comment.

File: pygmt/datasets/earth_relief.py [hunk 2]
```diff
@@ -144,12 +144,13 @@ def load_earth_relief(
     earth_relief_sources = {
         "igpp": "earth_relief_",
         "gebco": "earth_gebco_",
+        "gebcosi": "earth_gebcosi_",
         "synbath": "earth_synbath_",
```
Comment (weiji14) [inline pygmt/datasets/earth_relief.py:148]: Need to update the `data_source` parameter docstring at L71-L81 about the new `gebcosi` (and `synbath`) options.

**Review by weiji14 — COMMENTED** [review]
I think merging this PR first and then doing the refactor in #2200 will be better (good to have more unit tests in place before refactoring).

Oh, and can we close #1786 after this PR is merged?

File: pygmt/helpers/testing.py (hunk 1 above)
Comment (willschlitzer) [reply pygmt/helpers/testing.py:160]: Keep forgetting the `@` at the start; thanks!

File: pygmt/datasets/earth_relief.py (hunk 2 above)
Comment (willschlitzer) [reply pygmt/datasets/earth_relief.py:155]: Added!

**Review by weiji14 — APPROVED** [review]

**Pull request merged**